src/path_utils.rs

```diff
@@ -0,0 +1,36 @@
+use std::path::{Path, PathBuf};
+
+use path_slash::PathExt as _;
+use relative_path::RelativePath;
+
+use crate::error::WalkDirsError;
+
+pub fn get_slashed_path_buf<'a>(p: &'a Path) -> Result<PathBuf, WalkDirsError> {
+    let slashed: PathBuf;
+    #[cfg(target_os = "windows")]
+    {
+        slashed = match p.to_slash() {
+            Some(path) => PathBuf::from(path.to_string().as_str()),
+            None => return Err(WalkDirsError::SlashError(p.to_string_lossy().to_string())),
+        };
+    }
+    #[cfg(not(target_os = "windows"))]
+    {
+        slashed = PathBuf::from(p)
+    }
+    return Ok(slashed);
+}
+
+pub fn slashed_as_relative_path<'a>(
+    slashed: &'a PathBuf,
+) -> Result<&'a RelativePath, WalkDirsError> {
```

Comment:
I don't think this should be a reference here.

From the relativepath docs
https://docs.rs/relative-path/1.5.0/relative_path/struct.RelativePathBuf.html#method.from_path

The returned path looks like its own object that owns its own buffer holding its path, and has a lifetime that is totally independent from the input, so don't need to have a relationship between the result's lifetime and the original path's lifetime.

Reply:
I think you should be able to just return `Result<RelativePath, WalkDirsError>`, or if RelativePath can't be put in a Result because of dynamic size issues, you could box it with `Result<Box<RelativePath>, WalkDirsError>`

Reply:
Replaced the `RelativePath` with `RelativePathBuf` to avoid the box and reference thing 😬